back/app/services/invites/service.rb

```diff
@@ -0,0 +1,161 @@
+# frozen_string_literal: true
+
+require 'rubyXL'
+
+class Invites::Service
+  MAX_INVITES = 1000
+
+  delegate :add_error, :fail_now, to: :@error_storage
+
+  def initialize
+    @error_storage = Invites::ErrorStorage.new
+  end
+
+  def generate_token
+    [*('a'..'z'), *('0'..'9')].sample(9).join
+  end
+
+  def bulk_create_xlsx(xlsx_param, default_params = {}, inviter = nil)
+    hash_array, map_rows = Invites::XlsxProcessor.new(@error_storage).param_to_hash_array(xlsx_param)
+
+    bulk_create(hash_array, default_params, inviter)
+  rescue Invites::FailedError => e
+    e.errors.each do |error|
+      error.row && (error.row = (map_rows[error.row] + 2))
+      error.rows&.map! { |r| map_rows[r] + 2 }
```

Comment:
Why are we adding 2 here?

Reply:
I hope it's not important here. The file `invites_service` was renamed to `invites/service` hence the diff. The functionality should stay 100% the same.

Reply:
So, the words `invites` & `service` each use a row, for each error produced?

Reply:
Ignore me, I've just realised this is not your change, and is already in the codebase 🤦